examples/ipv6only-app/esp32/main/main.cpp

```diff
@@ -0,0 +1,620 @@
+            if (select_err < 0)
+                continue;
+
+            n = recvfrom(sockfd, buf, kUdpBufferSize, 0, reinterpret_cast<struct sockaddr *>(&clientaddr), &clientlen);
+            if (n < 0)
+                continue;
+            // Echo back
+            n = sendto(sockfd, buf, n, 0, reinterpret_cast<struct sockaddr *>(&clientaddr), clientlen);
+        }
+    }
+    // Never returns
+}
+
+} // namespace
+
+namespace chip {
+namespace rpc {
+
+class GDMWifiBase final : public generated::GDMWifiBase<GDMWifiBase>
+{
+public:
+    pw::Status GetChannel(ServerContext &, const chip_rpc_Empty & request, chip_rpc_Channel & response)
+    {
+        uint8_t channel = 0;
+        wifi_second_chan_t second;
+        PW_ESP_TRY(esp_wifi_get_channel(&channel, &second));
+        response.channel = channel;
+        return pw::OkStatus();
+    }
+
+    pw::Status GetSsid(ServerContext &, const chip_rpc_Empty & request, chip_rpc_Ssid & response)
+    {
+        wifi_config_t config;
+        PW_ESP_TRY(esp_wifi_get_config(ESP_IF_WIFI_STA, &config));
+        memcpy(response.ssid.bytes, config.sta.ssid, sizeof(response.ssid.bytes));
+        response.ssid.size = 32;
+        return pw::OkStatus();
+    }
+
+    pw::Status GetState(ServerContext &, const chip_rpc_Empty & request, chip_rpc_State & response)
+    {
+        wifi_ap_record_t ap_info;
+        esp_err_t err = esp_wifi_sta_get_ap_info(&ap_info);
+        PW_ESP_TRY(err);
+        response.connected = (err != ESP_ERR_WIFI_NOT_CONNECT);
+        return pw::OkStatus();
+    }
+
+    pw::Status GetMacAddress(ServerContext &, const chip_rpc_Empty & request, chip_rpc_MacAddress & response)
+    {
+        uint8_t mac[6];
+        PW_ESP_TRY(esp_wifi_get_mac(ESP_IF_WIFI_STA, mac));
+        sprintf(response.mac_address, "%02x:%02x:%02x:%02x:%02x:%02x", mac[0], mac[1], mac[2], mac[3], mac[4], mac[5]);
+        return pw::OkStatus();
+    }
+
+    pw::Status GetWiFiInterface(ServerContext &, const chip_rpc_Empty & request, chip_rpc_WiFiInterface & response)
+    {
+
+        wifi_ap_record_t ap_info;
+        PW_ESP_TRY(esp_wifi_sta_get_ap_info(&ap_info));
+        sprintf(response.interface, "STA");
+        return pw::OkStatus();
+    }
+
+    pw::Status GetIP4Address(ServerContext &, const chip_rpc_Empty & request, chip_rpc_IP4Address & response)
+    {
+        esp_netif_ip_info_t ip_info;
+        PW_ESP_TRY(esp_netif_get_ip_info(wifi_sta, &ip_info));
+        sprintf(response.address, IPSTR, IP2STR(&ip_info.ip));
+        return pw::OkStatus();
+    }
+
+    pw::Status GetIP6Address(ServerContext &, const chip_rpc_Empty & request, chip_rpc_IP6Address & response)
+    {
+        esp_ip6_addr_t ip6{ 0 };
+        PW_ESP_TRY(esp_netif_get_ip6_linklocal(wifi_sta, &ip6));
+        sprintf(response.address, IPV6STR, IPV62STR(ip6));
+        return pw::OkStatus();
+    }
+
+    void StartScan(ServerContext &, const chip_rpc_ScanConfig & request, ServerWriter<chip_rpc_ScanResults> & writer)
```

Comment:
Why does this take an lvalue reference only to move it?

Reply:
It takes the passed in RPC streamer and moves it to a global before returning, the global is used to stream the scan results. Hopefully more obvious now with the s_.

Reply:
Can it use a value or an rvalue reference parameter? Moving from lvalue reference is strange.

Reply:
@rgoliver ?

Reply:
Sorry for missing the reply to this comment, this was also cleaned up in the refactor. No longer gets moved to a global but is instead simplified to just use locally and return.